### PR TITLE
many fixes + some ✨

### PR DESCRIPTION
**Pull request by maxbeatty**
- [x] use GitHub profile to populate test and comment forms (no more local storage)
- [x] show logged in with GitHub
- [x] fix url used when creating revisions in browserscope
- [x] allow test to be deleted if its title and code are empty and updating own page
- [x] fix editing your own tests (no longer inserts new tests)
- [x] "My tests" link now works

### CODE REVIEW

File: server/repositories/browserscope.js [hunk 1]
```diff
@@ -2,13 +2,13 @@ const https = require('https');
...
 exports.register = function (server, options, next) {
-  server.expose('addTest', function (title, description, slug) {
+  server.expose('addTest', function (title, description, slug, rev) {
     return new Promise(function (resolve, reject) {
       var qs = querystring.stringify({
         'api_key': options.api_key,
         name: title,
         description: description.substr(0, 60),
-        url: options.scheme + '://' + options.domain + '/' + slug
+        url: `${options.scheme}://${options.domain}/${slug}/${rev}`
```
Comment (mathiasbynens) [inline server/repositories/browserscope.js]:
```
`${options.scheme}://${options.domain}/${slug}${rev ? '/' + rev : ''}`
```

File: test/e2e/home.js [hunk 2]
```diff
@@ -158,4 +159,13 @@ lab.experiment('Home page', () => {
...
     });
   });
+
+  /*
+    TODO: test additional flows
+
+    - edit page, add new test, save, verify page has 3 tests
+    - edit page, blank test title, blank test code, save, verify page has 2 tests
+    - edit page, blank test title, save, verify did not save and validation message exists on title
+    - edit page, blank test code, save, verify did not save and validation message exists on code
```
Comment (mathiasbynens) [inline test/e2e/home.js]: 👍

File: server/repositories/browserscope.js (hunk 1 above)
Comment (maxbeatty) [inline server/repositories/browserscope.js]: why shouldn't we always register the first revision?

I'd move this conditional logic out of this method so the callers are more explicit about what they are doing

Comment (mathiasbynens) [inline server/repositories/browserscope.js]: I originally designed the URLs this way because it felt cleaner. No big deal if this changes, though

**Comment by maxbeatty** [comment]
e2e tests are failing inconsistently on Saucelabs but not locally 😑

**Comment by maxbeatty** [comment]
@mathiasbynens @asilluron not sure how fruitful battling Saucelabs will be. if you're good with these changes, maybe we relax the required CI checks so this can get live and fix a lot of people's frustrations?

**Comment by asilluron** [comment]
For sure. I'm fine with that.

**Comment by mathiasbynens** [comment]
SGTM. Fixing the issues would be a great step forward, regardless of what Saucelabs thinks about this PR ;)

**Comment by mathiasbynens** [comment]
This is now live. Thanks, @maxbeatty! 👍

**Comment by mathiasbynens** [comment]
Hah, https://travis-ci.org/jsperf/jsperf.com/builds/194583931 passes 😕

**Pull request merged**